clang/lib/CIR/Dialect/Transforms/CIRSimplify.cpp

```diff
@@ -293,11 +169,7 @@ void CIRSimplifyPass::runOnOperation() {
   // Collect operations to apply patterns.
   SmallVector<Operation *, 16> ops;
   getOperation()->walk([&](Operation *op) {
-    // CastOp here is to perform a manual `fold` in
-    // applyOpPatternsAndFold
-    if (isa<BrOp, BrCondOp, ScopeOp, SwitchOp, CastOp, TryOp, UnaryOp,
```

Comment:
We can tackle this in followup PRs, but shouldn't `UnaryOp` be part of CIRSimplify? Some of its folding definitely affects source fidelity, perhaps we need to move some of that logic out of the `::fold` into a rewriter in CIRSimplify.